assembly/src/ast/parsers/context.rs

```diff
@@ -643,7 +645,13 @@ impl ParserContext<'_> {
             .get(module_name)
             .ok_or_else(|| ParsingError::procedure_module_not_imported(token, module_name))?;
         let proc_id = ProcedureId::from_name(proc_name, module_path);
-        Ok(proc_id)
+        match ProcedureName::try_from(proc_name.to_string()) {
+            Ok(name) => {
+                self.used_imported_procs.insert(proc_id, (name, module_path.clone()));
+                Ok(proc_id)
+            }
+            Err(err) => Err(ParsingError::invalid_proc_name(token, err)),
+        }
```

Comment:
I wonder if this is the right place to perform this check. Maybe a better way would be to pass `ProcedureName` as a parameter into this function - but I'm not sure how many things this would impact (would we just need to change `InvocationTarget` parsing?).

Reply:
Good idea. Let me try to see what the blast radius is.